Jpa integration

bmasles opened the pull request:
finished up the JPA integration and added the Eureka client dependencies

bmasles committed:
First version

bmasles committed:
Eureka

bmasles committed:
Had incorrect context path

bmasles committed:
isEmpty is not java 8

SanSanchez commented on src/main/java/com/smoothstack/lms/adminmicroservice/controller/AdminController.java:
```diff
@@ -34,6 +40,22 @@
+	@GetMapping(path = "/administrator/port")
+	public String port() {
+		return environment.getProperty("local.server.port");
+	}
```
Your controller isn't the right place for this kind of code. This should go in the main class or in a configuration file.

SanSanchez commented on src/main/java/com/smoothstack/lms/adminmicroservice/controller/AdminController.java:
```diff
@@ -363,11 +393,11 @@
+		branch.setId(id);
 		try {
 			adminService.updateBranch(branch);
 			return new ResponseEntity<>(HttpStatus.OK);
```
This try catch pattern is redundant, you're essentially just doing what the jvm was going to do for you anyway. You can remove the try catch block and make sure just not to suppress any exceptions raised in the service call.

SanSanchez reviewed (changes requested):
Overall good, I've made some comments. Please review them and revise your code appropriately. 

The only thing that concerns me (I didn't leave a comment for this.) is that you may not actually be properly doing unit testing. You're testing your application, but it doesn't capture the entire purpose of unit testing because it seems that you're not properly isolating the functions that you're unit testing. 

Also, don't include the bin file in the repository. It should be excluded. You can add the file pattern to your .gitignore file.

bmasles committed:
Added common, working on integration

still working on getting the new models and service integrated into the microservice

bmasles committed:
Common integration

bmasles committed:
Identity Provider Check
